tests/iotsdk/accesstoken.go

```diff
@@ -207,13 +207,23 @@ func verifyAccessTokenResponse(response thing.AccessTokenResponse, subject strin
 		anvil.DebugLogger.Println(err)
 		return false
 	}
+	type customClaims struct {
+		SubjectName string `json:"subname,omitempty"`
+	}
 	claims := &jwt.Claims{}
-	if err := accessJWT.UnsafeClaimsWithoutVerification(claims); err != nil {
+	custom := &customClaims{}
+	if err := accessJWT.UnsafeClaimsWithoutVerification(claims, custom); err != nil {
 		anvil.DebugLogger.Println(err)
 		return false
 	}
-	if claims.Subject != subject {
-		anvil.DebugLogger.Printf("access token subject, %s, not equal to thing ID, %s\n", claims.Subject, subject)
+	compoundSub := "(usr!"+subject+")"
+	if claims.Subject != subject && claims.Subject != compoundSub {
+		anvil.DebugLogger.Printf("access token sub, %s, not equal to thing ID, %s, or compound ID, %s\n",
+			claims.Subject, subject, compoundSub)
+		return false
+	}
+	if custom.SubjectName != "" && custom.SubjectName != subject {
```

Comment:
Is the `custom.SubjectName != ""` check redundant?

Reply:
Yes it is. I was checking for nil before and should have removed it.

Reply:
Actually, just went through the logic again and this is correct. If `custom.SubjectName` is empty, the statement should not evaluate to false.

Reply:
Yes indeed.